internal/journald/journald.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+/*
+Package journald provides functions to read and collect journald logs.
+*/
+package journald
+
+import (
+	"context"
+	"os/exec"
+)
+
+type collectionCommand interface {
+	executeCommand() ([]byte, error)
+}
+
+// Command represents a command that is executed by journalctl.
+type Command struct {
+	ctx     context.Context
+	service string
+	command *exec.Cmd
+}
+
+// NewCommand creates a new JournaldCommand.
+func NewCommand(ctx context.Context, service string) (*Command, error) {
+	cmd := exec.CommandContext(ctx, "journalctl", "-u", service)
+	if cmd.Err != nil {
+		return nil, cmd.Err
+	}
+	return &Command{ctx, service, cmd}, nil
+}
+
+// GetServiceLog gets all journald logs from a service and returns a byte array with the plain text logs.
+func GetServiceLog(jcmd collectionCommand) ([]byte, error) {
+	out, err := jcmd.executeCommand()
+	if err != nil {
```

Comment:
We need to check the error and, if it is an exit error, return the Stderr that command printed.
```suggestion
func (c *Collector) Collect() ([]byte, error) {
	out, err := c.cmd.Output()
	var exitErr *exec.ExitError
	if errors.As(err, &exitErr) {
		return nil, fmt.Errorf("executing %q failed: %s", c.cmd, exitErr.Stderr)
	} else if err != nil {
		return nil, fmt.Errorf("executing command: %w", err)
	}
	return out, nil
```